src/playlist-maker.js

```diff
@@ -0,0 +1,57 @@
+var playItem = require('./playitem.js');
+var setupAutoadvance = require('./autoadvance.js');
+
+var playlistMaker = function(player, plist) {
+  var currentIndex = 0;
+  var autoadvanceTimeout = null;
+
+  var playlist = function playlist(list) {
+    if (list) {
+      plist = list.slice();
+    }
```

Comment:
I might just be missing the obvious, but what's the scenario in which this occurs?